### PR TITLE
Add Duo Passcode support for both traditional and universal prompt

### LINKED ISSUE #162: Support for Passcode
In the latest version it's now available to choose auth method for DUO if default one not chosen. 
However, despite the fact that it now recognizes the Passcode method, it still works with push only. 
```
Sending request for authentication
Waiting for additional authentication
No default authentication method configured.
Please enter your desired authentication method (Ex: Duo Push): Passcode
Triggering authentication method: 'Passcode'
Exception in thread Thread-1:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/usr/lib/python2.7/threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/home/zhernovs/.local/lib/python2.7/site-packages/aws_adfs/_duo_authenticator.py", line 132, in _perform_authentication_transaction
    ssl_verification_enabled
  File "/home/zhernovs/.local/lib/python2.7/site-packages/aws_adfs/_duo_authenticator.py", line 565, in _begin_authentication_transaction
    u'Cannot begin authentication process. The error response: {}'.format(response.text)
ClickException: Cannot begin authentication process. The error response: {"message": "Please enter a passcode.", "stat": "FAIL"}
```

Please add functionality to enter passcode in that case. Some devices just don't support Duo-push.

### PR DESCRIPTION
**Pull request by trav-c**
* Added support for the Passcode method to both duo traditional and universal prompts
* Extended --duo-factor and --duo-device to support from duo universal prompt to also include duo traditional prompt
* Added pattern matching to make --duo-factor more forgiving (eg `--duo-factor WebAuthn` is now equivalent to `--duo-factor "WebAuthn Security Key"` and `--duo-factor Push` is equivalent to `--duo-factor "Duo Push"` etc).  Unmatched / unrecognized factors are still passed through as originally specified, and the original, full versions of the factor names are also still supported.

I have tested this with Passcode, Push, Phone Call and Webauthn for both the Universal and Traditional prompt, Passcodes work properly for both hardware (TOTP) tokens and Bypass codes, however as yet the 'sms' factor is not fully supported.  You can initiate an SMS (which will send the message, but then fail), then use the code(s) from the SMS message with the passcode method though.

This resolves #162

### CODE REVIEW

File: aws_adfs/login.py [hunk 1]
```diff
@@ -133,11 +134,11 @@
 )
 @click.option(
     "--duo-factor",
-    help=f'Use a specific Duo factor, overriding the default one configured server side. Known Duo factors that can be used with aws-adfs are "{DUO_UNIVERSAL_PROMPT_FACTOR_DUO_PUSH}", "{DUO_UNIVERSAL_PROMPT_FACTOR_PHONE_CALL}", and "{DUO_UNIVERSAL_PROMPT_FACTOR_WEBAUTHN}". Only supported with Duo Universal Prompt.',
+    help=f'Use a specific Duo factor, overriding the default one configured server side. Known Duo factors that can be used with aws-adfs are "{DUO_UNIVERSAL_PROMPT_FACTOR_DUO_PUSH}", "{DUO_UNIVERSAL_PROMPT_FACTOR_PHONE_CALL}", "{DUO_UNIVERSAL_PROMPT_FACTOR_WEBAUTHN}" and "{DUO_UNIVERSAL_PROMPT_FACTOR_PASSCODE}". Only supported with Duo Universal Prompt.',
```
Comment (pdecat) [inline aws_adfs/login.py:137]: Isn't the `Only supported with Duo Universal Prompt` part no longer valid with this PR too?
```suggestion
    help=f'Use a specific Duo factor, overriding the default one configured server side. Known Duo factors that can be used with aws-adfs are "{DUO_UNIVERSAL_PROMPT_FACTOR_DUO_PUSH}", "{DUO_UNIVERSAL_PROMPT_FACTOR_PHONE_CALL}", "{DUO_UNIVERSAL_PROMPT_FACTOR_WEBAUTHN}" and "{DUO_UNIVERSAL_PROMPT_FACTOR_PASSCODE}".',
```

**Review by pdecat — CHANGES_REQUESTED** [review]
Thanks for this awesome addition!

Left a few minor comments, but looks good to me!

**Comment by trav-c** [comment]
I've applied the proposed updates, not sure what happened with the help, since I did make that change, but must not have pushed it before opening my PR or something.  Otherwise noted on the ordering of the constants.

**Review by pdecat — APPROVED** [review]
LGTM, thanks again!

**Comment by pdecat** [comment]
Released in https://github.com/venth/aws-adfs/releases/tag/v2.6.0

**Comment by pdecat** [comment]
As I anticipated this PR also adds support for hardware tokens that generate HOTP codes (e.g. RSA).
I don't have one, but a colleague confirmed this to me. A PR is coming to explicitly mention this.

**Pull request merged**